### PR TITLE
Migration - assign missing pilLicenceNumbers to profiles

### PR DESCRIPTION
**Pull request by joefitter**
* check for profiles with null pilLicenceNumber and a pil which is not inactive or pending, generate licence numbers for these

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,6 +27,7 @@
   "homepage": "https://github.com/UKHomeOffice/asl-schema#readme",
   "dependencies": {
     "@asl/constants": "^0.7.1",
+    "crypto": "^1.0.1",
```
Comment (lennym) [inline package.json:30]: `crypto` is part of node core. You don't need this.

Comment (joefitter) [reply package.json:30]: derp

File: migrations/20200921121510_assign_missing_pil_licence_numbers.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+              })
+              .catch(() => {
+                console.error(`Error patching ${profile.id} with licence number ${licenceNumber}`);
+              });
+          });
+      }, Promise.resolve());
+    })
+    .catch(err => {
+      console.log(err);
+    });
```
Comment (lennym) [inline migrations/20200921121510_assign_missing_pil_licence_numbers.js:44]: Can we remove this `catch` block? I'd rather it failed.

Comment (joefitter) [reply migrations/20200921121510_assign_missing_pil_licence_numbers.js:41]: 👍

File: migrations/20200921121510_assign_missing_pil_licence_numbers.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+          .then(licenceNumber => {
+            console.log(`Assigning pilLicenceNumber ${licenceNumber} to profile ${profile.id}`);
+            return Promise.resolve()
+              .then(() => knex('profiles').where('id', profile.id).first().update('pil_licence_number', licenceNumber))
+              .then(() => {
+                console.log(`Successfully patched ${profile.id}`);
+              })
+              .catch(() => {
+                console.error(`Error patching ${profile.id} with licence number ${licenceNumber}`);
+              });
```
[the remark this answers is not in the document]
Comment (joefitter) [reply migrations/20200921121510_assign_missing_pil_licence_numbers.js:38]: 👍

**Pull request merged**